src/megaapi_impl.cpp

```diff
@@ -6614,6 +6622,38 @@ bool MegaApiImpl::isScheduleNotifiable()
     }
 }
 
+int MegaApiImpl::abortPendingActions(error preverror)
+{
+    while (!backupsMap.empty())
+    {
+        std::map<int,MegaBackupController*>::iterator it = backupsMap.begin();
+        delete it->second;
+        backupsMap.erase(it);
+    }
+
+    while (!requestMap.empty())
+    {
+        std::map<int,MegaRequestPrivate*>::iterator it = requestMap.begin();
+        if(it->second)
+        {
+            fireOnRequestFinish(it->second, MegaError(preverror ? preverror : MegaError::API_EACCESS));
+        }
+    }
+
+    while (!transferMap.empty())
+    {
+        std::map<int, MegaTransferPrivate *>::reverse_iterator it = transferMap.rbegin();
+        if (it->second)
```

Comment:
if it->second is null, it will loop forever?

Reply:
Same as above